tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -436,3 +437,230 @@ def test_consumer_groups(self):
         for ex in self._exceptions:
             if ex:
                 raise ex
+
+
+# Original listdir in RedpandaService uses SFTP underneath.
+# Unfortunately, ducky hangs when calling SFTP procedures
+# while producers are running.
+# Therefore, the below is a re-write of listdir using SSH
+# only.
```

Comment:
nit: more investigation is needed to determine if the problem is on our end or on ducktape's side but this shouldn't block the PR.